### ISSUE TITLE
Review Content Types and What They Mean

### ISSUE THREAD

**Issue by markcmiller86** [betterscientificsoftware/betterscientificsoftware.github.io#361]
### BSSw Sub-Comittee Homework Assignment Due May 30, 2019
Before we continue developing editorial *workflow* for various types of content, we should probably review the content types we have, decide if we need to adjust the set of types we wanna support and also identify what criteria we use to determine what a given content type is or any specific requirements it may have.

This issue is about having that conversation.

Our current content types are described in our [What To Contribute as of May 23, 2019](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/91648f8f992639ad8a9f5467e00cf6dc1bec21a7/WhatToContribute.md) document. So, lets go review that first. After revewing that, add comments/discussion here about this. One key question that has come up recently is whether we have a few different types of *articles* and not just *blog* articles.

When you have finished with your commentary, etc. on this topic, please check your name off the following list...

- [ ] @rinkug 
- [x] @markcmiller86 
- [x] @pagrubel 
- [ ] @prwolfe 
- [ ] @curfman 
- [ ] @bernhold 
- [x] @bartlettroscoe 
- [ ] @JeanShuler

**Comment by bartlettroscoe** [comment]
To pin this down to a specific version and not just 'master' (which can change):

* [Snapshot of WhatToContribute.md as of 5/22/2019](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/91648f8f992639ad8a9f5467e00cf6dc1bec21a7/WhatToContribute.md)

**Comment by markcmiller86** [comment]
Sure, but lets do that by editing original comment

**Comment by bernhold** [comment]
I propose that **Events** should be treated as similar to **Curated Content**, at "worst".  I think arguments could be made for an even lighter editorial review process.

**Comment by bernhold** [comment]
Elsewhere (not on the cites page), I've been using the term **original article**, which I would propose encompasses the terms **original experience**, **whatis document** and **howto document** in the reference document.  I think they can can be defined similarly, in a general sense, and subjected to a similar editorial process.

**Comment by bartlettroscoe** [comment]
So we have:

* **Original experience:** An original article (brevity is appreciated) to inform the CSE community about how to improve developer productivity and software sustainability.
* **Blog article:** An original article in the form of a blog of 250 - 500 words. We will solicit contributions from thought leaders in the community and welcome proposals from anyone.

So should all "original experiences about how to improve developer productivity and software sustainability" not candidates for blog articles?   Or is anything that describes "how to improve developer productivity and software sustainability" not a candidate for a blog article?  Seems like every bit of content on the BSSw should help to "improve developer productivity and software sustainability".

Can an "original experience" be as long as you want?  Seems like we should have some general length guidelines for each of these.  I know previously we wanted "What Is" and "How To" documents to be shorter, shooting for about 2 pages of a formatted PDF document.

**Comment by bernhold** [comment]
I propose that **blog articles** and **original articles** be treated separately, at least for the time being.

I think sourcing is one key differentiator. I'm not sure whether there are other differentiators (in content or or in editorial process).

**Comment by bernhold** [comment]
> So we have:
> 
> * **Original experience:** An original article (brevity is appreciated) to inform the CSE community about how to improve developer productivity and software sustainability.
> * **Blog article:** An original article in the form of a blog of 250 - 500 words. We will solicit contributions from thought leaders in the community and welcome proposals from anyone.
> 
> So should all "original experiences about how to improve developer productivity and software sustainability" not candidates for blog articles? Or is anything that describes "how to improve developer productivity and software sustainability" not a candidate for a blog article? Seems like every bit of content on the BSSw should help to "improve developer productivity and software sustainability".
> 
> Can an "original experience" be as long as you want? Seems like we should have some general length guidelines for each of these. I know previously we wanted "What Is" and "How To" documents to be shorter, shooting for about 2 pages of a formatted PDF document.

I suggest that we not get too hung up on length for **blog articles** and **original articles**.  @bartlettroscoe did word counts on existing blog articles and found that reality is typically significantly longer than the 250-500 words mentioned for blogs.

I think it is reasonable to expect that **curated content** and **events** are fairly concise, and other types are likely to be longer.

In my mind, I think the upper bound for length is about whether it makes sense for a document to be a standalone paper, report, or other publication. (Which, by the way, is probably easier for the author to get recognition for.)

**Comment by prwolfe** [comment]
I like the "Guidelines" but I agree that the Types are a bit overlapping particularly the Original Experience, and Blog Article. Seems like a lot of this can be after the fact evaluations by the editorial board - get content and then decide where it fits. I agree with the idea of using the same review process as much as we can for all the types as I don't see a lot in that process that is type specific.

**Comment by markcmiller86** [comment]
I thought **original experience** was effectively an *objective, experience report* by some practitioner who had used a given technology in depth (not trivially) and had some concrete, objective observations to record about the experience cast, hopefully, as a series of advantages and disadvantages or strenghts and weaknesses.

For example, maybe someone attempts to use Raja or Kokkos for performance portability in an existing application. They do quite a bit of work and have something useful to say as a result of the experience.

An article of this type demands careful review because early drafts are likely to contain strongly worded opinions which then detract from the main purpose of the article to provide objective observations about how using a technology went.

I would like to not loose this important distinction about this type of content. That said, maybe qualities or attributes such as this don't need to be codified in how we name the content and can instead be codified by YAML base matter *tags*. So, for example, when as @bernhold proposes, an **Original Article** is drafted and that article contains content of this *experience* nature, then it gets tagged with `Judgy: True` or something to that effect.

**Comment by bernhold** [comment]
My feeling is that maybe there are two tiers in terms of the level of review that would be reasonable

  *   **curated content** and **events** would be a minimal, “sanity check” level of review.  This is justified because they’re short and pretty straightforward.
  *   **blog articles** and **original articles** may warrant a more thorough review.  @markmiller86 commented on some of the issues here.

**Comment by bernhold** [comment]
> I thought **original experience** was effectively an _objective, experience report_ by some practitioner who had used a given technology in depth (not trivially) and had some concrete, objective observations to record about the experience cast, hopefully, as a series of advantages and disadvantages or strenghts and weaknesses.
> 
> For example, maybe someone attempts to use Raja or Kokkos for performance portability in an existing application. They do quite a bit of work and have something useful to say as a result of the experience.
> 
> An article of this type demands careful review because early drafts are likely to contain strongly worded opinions which then detract from the main purpose of the article to provide objective observations about how using a technology went.
> 
> I would like to not loose this important distinction about this type of content. That said, maybe qualities or attributes such as this don't need to be codified in how we name the content and can instead be codified by YAML base matter _tags_. So, for example, when as @bernhold proposes, an **Original Article** is drafted and that article contains content of this _experience_ nature, then it gets tagged with `Judgy: True` or something to that effect.

I think any **original article** has the potential for more controversy, but we also want to maintain certain quality standards. For example, WhatIs and HowTos may not be so controversial (maybe I'm wrong about that), but we still want them to be high quality.  So peer review, similarly to how a journal or conference would approach it seems useful here.  And also to check that articles are doing what we consider approrpriate w.r.t. opnions and such.

**Comment by markcmiller86** [comment]
I see sort of two thinking caps being worn here. One is what the differences are in *workflow* for one type of content vs. another. It may be that a lot of different content types wind up mapping to the same *workflow*. That doesn't necessarily mean maintaining a distinction in those content types for other purposes isn't needed.

For example, *What Is* articles have an important role in the site, almost like parts of a glossary and text from them might be used to banner or otherwise decorate certain pages generated on the site. In addition, those items might be *auto-linked* wherever the term they define is used throughout the site, etc. So, even if *What Is* articles follow the same *workflow* as other types of content, they should not loose this distinction.

Again, however, its concievable that this distinction is maintained not so much by naming a content *type* but instead tagging it in YAML base matter. Probably the back-end IT support team has good reasons for wanting to distinguish content using one method (naming) vs another (YAML tags). We should ask them.

**Comment by markcmiller86** [comment]
> I think sourcing is one key differentiator.

So, I am not sure I understand what you mean by *sourcing* here. You mean *who* authors the content?

**Comment by markcmiller86** [comment]
I think *events* have some very specific content...the 5 W's, right? Who, What, Where, When, Why and maybe How (for how to attend such as registration or reservation link). *When* probably has date, time and timezone (can we include add-to-calendar helpers for Outlook, Gooble Calendar, etc.). *Where* might be a virtual or actual place.

But, I think events should be structured this way, with links to more information if useful/needed, to keep them concise, as @bernhold recommends, and easy/standard format to quickly read.

**Comment by markcmiller86** [comment]
I just wanna re-iterate...we should be in touch with the IT support team for the backend to ensure we understand their needs in terms of content type distinctions and what they mean for backend processing, etc.

**Comment by bernhold** [comment]
> > I think sourcing is one key differentiator.
> 
> So, I am not sure I understand what you mean by _sourcing_ here. You mean _who_ authors the content?

Historically, for blogs, Lois and I actively solicit articles from specific people.  There are few that appear unsolicited.  

Whereas for other types of content, it has been much more a matter of waiting for things to show up. (Yes, sometimes we, er, encourage, the IDEAS team to contribute to help grow the site.)

My guess is that **original articles** are likely to be a mix of contributions and organized campaigns (i.e. a new or updated round of HowTo and WhatIs documents) which, ideally, are "community" affairs.

So those are different "sourcing" scenarios that I'm referring to.

**Comment by markcmiller86** [comment]
> I propose that **Events** should be treated as similar to **Curated Content**, at "worst". I think arguments could be made for an even lighter editorial review process.

I am not sure we need PRs and Kanbahn to manage events. I think those can be handled by a very specific submission process (maybe a google form or maybe just a markdown issue template), plus maybe a ci check to root out bad submissions. We need the barrier to entry to be really low for this if we're going to gather inputs from a lot of sources. We don't wanna blindly go live with any ole' event someone contributes but as close to blindly as possible seems like the right goal for an *events workflow*, which I think rules out use the same workflow as **Curated Content**

**Comment by bernhold** [comment]
> I just wanna re-iterate...we should be in touch with the IT support team for the backend to ensure we understand their needs in terms of content type distinctions and what they mean for backend processing, etc.

There are definitely differences right now in how documents are processed depending on which top-level directory of the GitHub they're in (approximately corresponding to the content types we're discussing). Unfortunately, they're not documented (a request of long standing).

But let's not let the tail wag the dog -- if we need things to be handled differently, we should ask for that.  We've done it in the past.

**Comment by bartlettroscoe** [comment]
@markcmiller86 said:

> It may be that a lot of different content types wind up mapping to the same workflow. 

I would recommend using the simplest identical workflow for everything to start with to make this simple.  Create issue, create PR, have PR reviewed and approved, merge PR.

**Comment by markcmiller86** [comment]
> I would recommend using the simplest identical workflow for everything to start with to make this simple. Create issue, create PR, have PR reviewed and approved, merge PR.

Sure, we can do that if it makes sense to start with. I was just trying to emphasize that workflow for a given content type is, IMHO, not a distinguishing feature of different content types, whether or not they follow the same or different workflows.

**Comment by markcmiller86** [comment]
> But let's not let the tail wag the dog -- if we need things to be handled differently, we should ask for that. We've done it in the past.

I agree. But, my comment about engaging backend team on questions here wasn't so much to have them help us decide what content types we should support or what those types should mean but to let us know if tagging via YAML matter (or other means) vs. *where* the markdown files are placed within some folder hierarchy vs. file naming conventions have any advantages/disadantages for them. Right now, we appear to be using either the latter or maybe a hybrid of these approaches. 

Events seem different enough from other content, just putting them in their own folder probably makes sense. This might also be true for Curated Content too. But for the broader category of *articles* (e.g. blog posts, original articles as @bernhold proposed, etc.), a folder or folder hierarchy makes less sense, to me anyways, than a *tagging* approach.

In Jekyll speak (the reason I mention Jekyll is because I am also thinking longer term where we may not have the resources to pay for a back-end IT staff to continue supporthing the site), Events, Curated Content and Articles could be their own *collections* in the site. Articles could be further sliced by *category* tagging.

Some of the choices we make here about the mechanics of *identifying content types* does impact current back-end effort as well as our longer term sustainability.

**Comment by markcmiller86** [comment]
I propose [this PR](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/pull/364) for an issue template for events and the *Event* content type itself.

What I am hoping for is that **Anyone** can submit an event via this template and upon approval from **EB Member**, we can go from the markdown in the issue to the *published* form(s) of the event.

I propose that event workflow be basically...
1. **Anyone** submits issue
1. A single **EB Board** member (whoever jumps on it) who approves of the event, runs script that takes the markdown from the issue and produces the published form(s) of the event.

In particular, no PRs, no Kahnban board, etc.

**Comment by markcmiller86** [comment]
I propose that one thing we (well **EB Members** anyways) need to do is meet, probably every other week for half-hour, to process *new* issues since last meeting.

I think this is a key activity and precursor step to *any* workflow for content that would follow

**Comment by bernhold** [comment]
> I propose [this PR](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/pull/364) for an issue template for events and the _Event_ content type itself.
> 
> What I am hoping for is that **Anyone** can submit an event via this template and upon approval from **EB Member**, we can go from the markdown in the issue to the _published_ form(s) of the event.
> 
> I propose that event workflow be basically...
> 
> 1. **Anyone** submits issue
> 2. A single **EB Board** member (whoever jumps on it) who approves of the event, runs script that takes the markdown from the issue and produces the published form(s) of the event.
> 
> In particular, no PRs, no Kahnban board, etc.

The template needs some work to align with what we currently have in Events on the site.  I will work on that separately.

But I support the underlying suggestion from @markcmiller86 that **Events** could be okay with an even simpler workflow than **Curated Content**.

**Comment by markcmiller86** [comment]
> For example, _What Is_ articles have an important role in the site, almost like parts of a glossary and text from them might be used to banner or otherwise decorate certain pages generated on the site.

Looks like I am completely wrong about this characterization of the **WhatIs** content type. **WhatIs** documents are actually quite long and cannot be used for *decoration* on pages or as banner headings because of their length. However, their quality of serving as a place to link wherever the associated topic is mentioned is probably still true/relevant.

My recollection of having to come up with *very short* definition-like headings for a number of topical areas I think is from the recent site-redesign effort we engaged in.

Having said all that, I think it makes a lot of sense to *require* that the **WhatIs** content type include a short (character or word length limited) *definition* along with whatever other descriptive/instructive text is provided. We could probably audit all our **WhatIs** articles to identify the one sentence in them that would serve that role and either tag it somehow or bring it out to the top of the document.

**Comment by markcmiller86** [comment]
Maybe it is helpful to look at this problem by examining what are some of the *attributes* of content that might help to distinguish different content *types*? Let me just throw out here some possible attributes to stimulate thinking...

* Length limited
  * limiting content length has certain advantages that impact how the content can be used
  * tweets, page banners, cognitive load for readers
  * **Events**, **WhatIs definitions** (see @markcmiller86 comments above), *did-you-know* kinds of things seem appropriate for length limiting
* Contains Package/Product evaluative commentary
  * This kind of content requires *extra special* editorial review
  * **Original Experience** I thought had this attribute
  * Might require more **EB Members** to approve than normal
* Time sensitive
  * Certainly events are time sensitive. Any other kinds of content?
  * Maybe stuff that is likely to stale, fall in to disuse, is known to have limited life
  * Announcements of key infrastructure changes (e.g. Python 2 no longer supported after 2020, OSX requiring signed/keyed binaries)
* *Media Modality*
  * Tweets, text, videos, Jupyter notebooks, ...
* Fully vetted
  * Journal article (we might not be there yet) vs. opinion piece
* Is or contains Advertising
  * Will we eventually do this like any paper publication? I hope note but...
  * What if a vendor wants to share something about a new product  release?
* Historical
  * Kinda like the Apollo Guidance Computer article
  * Historical for curiosity's sake or historical as potential to inform current thinking

As I re-read this impromptu list, these attributes might be useful in *some* way but I am not sure I see a lot useful here in helping to slice up the *Content Types* pie we are aiming at here.

Maybe, what is *most important* regarding a *Content Type* is how much and what kind of *oversight* or *baby sitting* or whatever it requires by **EB Members**.  I guess that more than anything else kinda sorta establishes the *workflow* that will be followed to finally arrive at a published/live bit of content. If we think in these terms, we have a different list of *attributes*

* Number of **EB Member** approvals
* Requires branch and PR
* Requires only issue submission
  * I don't know if many of us recognize this but issues themselves are markdown documents which can be edited (though they are not version controlled). So, content authors could, in theory, do all their document editing right here in an issue without adding a new file to the repo, doing a commit or pull request or whatever. For a lot of the documents in our current inventory, a great number of them could have been easily authored in this way (and you get quick preview tab while doing it to boot).
* Requires editorial review by subject matter expert(s)
* Requires editorial review for evaluative commentary
* Requires finding an author and getting a commitment to create the content by a certain date

In my mind, each of the above might represent different or additional steps in, perhaps, the *one and only* workflow we use rather than wholly different workflows.

**Comment by markcmiller86** [comment]
Well, GitHub is doing something silly with email replies like [this one](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/issues/361#issuecomment-495340362). 

- First, they wind up including a *ton* of email metadata which is worthless and distracting.
- Next, they don't support markdown or @mentions 

Look at the picture below, screen-clipped from my attempt to edit @bernhold email to format it better, lower-left corner message. I've looked in vein for a setting that controls this and can't find anything. I will try some email test responses

<img width="914" alt="Untitled" src="https://user-images.githubusercontent.com/5720676/58376846-c4484200-7f28-11e9-892c-e3ccaf2dbb13.png">

**Comment by markcmiller86** [comment]
Test reply.

**Comment by markcmiller86** [comment]
Test reply.

Ok, but do you have to test it so much (note to self).

**Comment by markcmiller86** [comment]
Plain text reply

Does @markcmiller86 see this?

> This is a quote

### This is a heading

**This is bold text**

**Comment by markcmiller86** [comment]
> *Incidentally, this comment was produced by replying to the email notification. I wanted to see how it would come out. Not as pretty as doing it directly.*

@bernhold follow these steps for improved email reply behavior (see [above example](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/issues/361#issuecomment-495964816))

1. reply in plain text
1. Manually cull all boilerplate text (top/bottom) in email before drafting reply
1. Add markdown quote char (>) ahead of lines from others you are replying too
1. @mentions still work (just be sure to spell github handle correctly -- a mistake and likely some other github user will get included in the conversation). Note also that Outlook as its own @mention completion which could easily get in the way of this.

Not very satisfying but you get better results here if you can follow these steps.

**Comment by bernhold** [comment]
@markcmiller86 wrote:
> Maybe it is helpful to look at this problem by examining what are some of the _attributes_ of content that might help to distinguish different content _types_? Let me just throw out here some possible attributes to stimulate thinking...
> ...

> As I re-read this impromptu list, these attributes might be useful in _some_ way but I am not sure I see a lot useful here in helping to slice up the _Content Types_ pie we are aiming at here.

I'm inclined to agree that most of @markcmiller86's "attributes" are not what I'd use as differentiators.  Though there are a few things that I think would appear only in certain types of content:
* Contains Package/Product evaluative commentary -- I think this immediately kicks up the complexity. I would expect to see this kind of content only in **original articles** or **blog articles**
* Fully vetted -- I don't think we want to hold ourselves out as performing academic-style peer review.  I think our review processes are intended to provide a more basic level of quality control.  If something has already gone through academic peer review, I think we treat it as **curated content**.  I can see **original articles** and perhaps **blog articles** *referring* to fully vetted resources, not *being* them.
* Advertising -- personally, I don't think that's appropriate for BSSw, though I recognize that there is sometimes a fine line as to what constitutes advertising.  Consider a blog post about reproducibility which touts a particular tool.  What if that tool happens to be a commercial product?
  - We need to separate this from the idea of using some of the space on the site for advertising (as many blogs and other sites do). This would not be BSSw content, but rather renting out real estate on the site, and customer eyeballs.  We may not want to go this direction either, but we must recognize that it could be a source of (probably modest) income in a post-ECP world.
* Historical -- I would only expect to see such content in **original articles** or **blog articles**

My feeling is that the editorial processes we need have to do with the *complexity* and *originality* of the material, together with the *quality threshold* we want to apply and the *time sensitivity*. Here's a simple set of ratings in *my* opinion:

| Content Type|  Complexity | Originality | Quality Threshold | Time Sensitivity |
|------------------|---------------|---------------|-------------------------|----------------------|
| Event | L | L | L | H |
| Curated Content |  M |  L-M | M | L |
| Original Article |  M-H | H | M-H | L |
| Blog Article | M-H | H | H | H |

Note that *time sensitivity* arises in two ways: **events** are time sensitive because if they expire on a certain date, and the sooner they're on the site, the more useful they are.  **Blog articles** are time sensitive because we have (artificially imposed on ourselves) a particular publication schedule (middle and end of the month) and we need processes structure to help meet those goals.

**Comment by markcmiller86** [comment]
I am a bit surprised to see *originality* in your list of attributes here. What does *original* mean in this context? Not published anywhere before? Not published in CSE/HPC community before? Not published on BSSw before?

> * I don't think we want to hold ourselves out as performing academic-style peer review.

So, I had't considered the implications of this and maybe *fully* and/or *vetted* isn't the right verbiage for what I was proposing there. I do think we'll have situations where **EB Members** are not sufficiently informed to render an opinion on lets sey *originality* for example and we we might need to enlist the opinion(s) of (volunteered) SMEs. Thats kinda sorta what I mean by *vetting*.

**Comment by bernhold** [comment]
> I am a bit surprised to see _originality_ in your list of attributes here. What does _original_ mean in this context? Not published anywhere before? Not published in CSE/HPC community before? Not published on BSSw before?

Fair question.  I guess that I mean: **the extent to which the content of interest to our community resides on BSSw vs other sites.**  The extent to which the content is "original" to us.

So I perceive **events** as having low originality -- the main purpose is to let the community know about something, but the content of real interest is on the event's web site.

I think most **curated content** is going to be low originality too, because the interesting material, by our definition, resides elsewhere.  We're primarily providing a pointer, with a little bit more to say why someone might be interested in the resource.

Now some ambitious contributors will take **curated content** a step further and provide more extensive information.  For example, @bartlettroscoe's summary of Legacy Code (#331) on the assumption that few will actually want to read the whole book.  To me, that's a higher level of originality, and for that reason, probably deserves a bit of a closer look than a "standard" **curated content** article -- more of the content of interest to readers is on our site, as opposed to the book.

My expectation is that **original articles** and **blog articles** are generally going to have the interesting content on the site, relying less on outside material.

That said, there are a couple of corner cases I can see for **blog articles**:
- One is reposting material that's already appeared elsewhere. For example, https://bssw.io/blog_posts/software-as-craft was originally published on Paul's LinkedIn page. But since we're hosting the content rather than merely providing a curated content-style link to an outside resource, I think this would be considered to have the primary content of interest on the BSSw site (it just happens that there's another copy hosted elsewhere).
- Fairly often, we use the blog to draw attention to things -- events (future or past), the BSSw Fellowship competition and results, etc.  In these cases, I think it is reasonable to grade them lower on "originality", I don't see that as a negative -- to me it merely says that such articles probably don't need as intensive review (though they still need to be reviewed).

**Comment by bernhold** [comment]
> > * I don't think we want to hold ourselves out as performing academic-style peer review.
> 
> So, I had't considered the implications of this and maybe _fully_ and/or _vetted_ isn't the right verbiage for what I was proposing there. I do think we'll have situations where **EB Members** are not sufficiently informed to render an opinion on lets sey _originality_ for example and we we might need to enlist the opinion(s) of (volunteered) SMEs. Thats kinda sorta what I mean by _vetting_.

Gotcha.  I would characterize this situation differently.  I don't see it as being about the *level* of vetting (e.g., full, medium, minimal), but rather a question of *who* we solicit for reviews.  

For a given level of vetting that is needed for a certain type of content or specific article, can we comfortably handle it within the **EB** or do we need to engage outside **SME**s?

I think we *should* expect to vet different content types at different levels.  But personally, I feel that we should expect any generalist in SWE for CSE to provide adequate levels of vetting for most of the content we publish on the site -- the threshold simply should not be all that high.

To me, only for a subset of articles -- the most complex and/or original -- should we expect to review at a deep enough level that we have to worry about whether the needed expertise is available on the **EB** or not.

**Comment by markcmiller86** [comment]
> we should expect any generalist in SWE for CSE to provide adequate levels of vetting for most of the content we publish on the site -- the threshold simply should not be all that high

I agree with that. Seems like the right level to be aiming for too.

**Comment by markcmiller86** [comment]
> That said, there are a couple of corner cases I can see for **blog articles**:
> 
> * One is reposting material that's already appeared elsewhere.

So, I have a similar example of that. [This article](https://bssw.io/blog_posts/continuous-technology-refreshment-an-introduction-using-recent-tech-refresh-experiences-on-visit) on BSSw conceptually originated from [this article](https://computation.llnl.gov/newsroom/great-migration-visit-moves-subversion-github) on LLNL's Computation site.

However, the BSSw article represents, IMHO, some significant originality relative to the LLNL article because it

* casts the discussion in terms of *tech refresh* and indeed defines the term *continuous technology refreshment (CTR)* for CSE/HPC community
* goes into specific details about the technologies involved and how to use them to achieve the specified goals
* links to various useful resources

So, even though the basic idea appeared eleswhere before, I regard the article here on BSSw to be pretty *original*, using your characterization of that attribute.

**Comment by markcmiller86** [comment]
I wonder if the **Event** Content Type should really be **Events and/or Announcements** or maybe we should just agree that the **Event** Content Type can be used for *events* or *announcement*.

An example of an announcement...Machine selections, Major program starts (e.g. Euro-ECP), KNL retiring, Python 2 sunsetting,

**Comment by bernhold** [comment]
> I wonder if the **Event** Content Type should really be **Events and/or Announcements** or maybe we should just agree that the **Event** Content Type can be used for _events_ or _announcement_.
> 
> An example of an announcement...Machine selections, Major program starts (e.g. Euro-ECP), KNL retiring, Python 2 sunsetting,

As far as terminology goes, bssw.io already has something called **announcements**.  That's the material that appears from time to time in the yellow band near the top of the page (until you dismiss it).  Traditionally, we've used it to announce the posting of new blog articles.  I think it was originally limited to just one item.  Then we we got it expanded to two.  And I *believe* that in the next incarnation, it will be general to however many we want.

I'm happy to agree that the **event** content type could really be an announcement as you're using the term. We already do use it to announce some things, like calls for papers or journal special issues.

But then, as always, we want to be thoughtful about what events or announcements we consider appropriate to publish on bssw.io.

For example, we publish calls for papers for events that are related to SWE for CSE.  For example, identifiable events being held under the umbrella of conferences like CSE, SC, ISC, PASC and other.  But we don't publish general events for those conferences.

I'm not going to read a huge amount to the particular list of things you gave as examples of possible announcements, but here's my *personal* reaction to them:
* In scope: funding programs that have an SWE component, Python 2 sunsetting
* Out of scope: machine selections, most funding announcements, KNL retiring

Of course I'm open to discussion of this classification.

**Comment by markcmiller86** [comment]
> * Out of scope: machine selections, most funding announcements, KNL retiring

I guess I am thinking any announcement that is likely to impact HPC/CSE software development planning is something to consider *in scope*. OTOH, machine selections are likely fully covered elsewhere as are funding announcements. So, maybe criteria ought to be *likely to impact HPC/CSE software development planning* **and** *is less likely to have been fully covered elsewhere*.

**Comment by bernhold** [comment]
> So, maybe criteria ought to be _likely to impact HPC/CSE software development planning_ **and** _is less likely to have been fully covered elsewhere_.

My boss's standard career advice is "only *do* what only *you* can do". (Emphasis added to aid parsing.)  I think that advice also applies to BSSw.  And I think your criteria is consistent with that.

**Comment by jarrah42** [comment]
Ok, I've waded through all 40+ comments and I actually think this document is the wrong one for an editorial workflow and that we need a separate document. It seems to me that it describes the content we are interested in to an audience of potential authors (which I'm not saying we don't need, and is fine for that purpose), however it is largely irrelevant to the types of workflow that we need to publish a document. For the latter, there is really only one primary consideration: is the content created by a third party or not? If not, then we already have a workflow for curated content which would work equally well for other documents that are created by the EB. If a third part is involved then the author needs to be involved in the review process and we need a workflow that encompasses that, such as that used by JOSS. It seems like the curated content workflow could be a good starting point for this as well, and we could just add some additional steps in order to provide feedback and get updates from the author. Anyway that's my 2c worth.

**Comment by markcmiller86** [comment]
> Ok, I've waded through all 40+ comments and I actually think this document is the wrong one for an editorial workflow and that we need a separate document.

@jarrah42 ... Just to be clear, there is no *document* here. This *issue* is just to capture conversation about out content types and to, perhaps, hopefully, achieve a consensus on what the types are and how they are distinguished from each other. Once we understand that, then talking about workflow would follow.

> there is really only one primary consideration: is the content created by a third party or not?

Potentially, this is equivalent to @bernhold concept of the *originality* attribute.

> If not, then we already have a workflow for curated content which would work equally well for other documents that are created by the EB.

Hmmm...just because some content is of type **Curated Content** doesn't mean it is or is not written by a third party. It seems like you are refering to the *quality* of a **Curated Content** article that its primary purpose is to refer to some pre-existing content, somewhere else, that we don't have to develop. That is mostly true but have a look at [this PR](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/pull/359) by @bartlettroscoe. There is a lot *original* work by @bartlettroscoe there to help cull out key points just in case the reader doesn't have easy access to the book or wanna read the *whole* thing to get the key points. In fact, that might even be a *best practice* for BSSw curated content type; to cull out in the curated content article, some key points as opposed to simply referring readers to the curated stuff.

**Comment by jarrah42** [comment]
> @jarrah42 ... Just to be clear, there is no _document_ here. This _issue_ is just to capture conversation about out content types and to, perhaps, hopefully, achieve a consensus on what the types are and how they are distinguished from each other. Once we understand that, then talking about workflow would follow.

I thought it was fairly obvious, but in any case I'm referring to the document you referenced in the first comment: [What To Contribute as of May 23, 2019](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/91648f8f992639ad8a9f5467e00cf6dc1bec21a7/WhatToContribute.md) 

> > If not, then we already have a workflow for curated content which would work equally well for other documents that are created by the EB.
> 
> Hmmm...just because some content is of type **Curated Content** doesn't mean it is or is not written by a third party. It seems like you are refering to the _quality_ of a **Curated Content** article that its primary purpose is to refer to some pre-existing content, somewhere else, that we don't have to develop. 

I would think that in most cases the curated content will just be a short description of what the other content is, and it seems unlikely that third parties (aka non-EB members) would have much interest in writing it (I could be wrong, but has anyone done it to date?). Even if they did, once the CC idea has been approved it seems like the actual article would be fairly simple and non-controversial. This is in contrast to an article or blog which an author would have a vested interested in, and which would require a more intensive review process. It seems like these two are distinct enough to warrant different workflows, but I don't see that "what is" or "original experience" documents are going to require their own workflows, but would use the same workflow as a blog. 

In other words, there are likely to be only a couple of different workflows regardless of the number of different content types we come up with. So if it this issue is about determining editorial workflows, then we've probably already got enough examples to figure out what workflows we need.

**Comment by bernhold** [comment]
> I would think that in most cases the curated content will just be a short description of what the other content is, and it seems unlikely that third parties (aka non-EB members) would have much interest in writing it (I could be wrong, but has anyone done it to date?).

First of all, yes, we already have resources (besides blog articles and events) which have been contributed by outsiders (to the EB, to the IDEAS project).  See the [Contributors page](https://bssw.io/items/authors/).

Second, I think it is undesirable and even inappropriate for the bulk of the content in what is intended to be a *community* resource to come from a small handle of people (the EB).  What will make this a useful resource is if it represents a diversity of experience and what people find beneficial.

Personally, I will consider BSSw to be a failure if we can't start attracting non-trivial outside contributions.  I'm willing to give it some time, but I have my eye on a very different asymptote.

**Comment by bernhold** [comment]
@jarrah42, I do want to thank you very much for wading through the discussions and taking the time to provide your thoughts.  Until now, the conversation has been exclusively between @markcmiller86 and myself.  And while it has been fun, I think we would benefit from additional perspectives.

**Comment by pagrubel** [comment]
Having read through all of this conversation and looking at WhatToContribute.md
I really can't add all that much except I agree Events should be fairly easy to get on the site without too much process.
Why not put What Is and How To articles in subfolders? Seems to make it a little clearer. I'd even like to see those clearer on the actual site, but I have been gone and haven't gotten to look at it yet (next on my list).
I'll look back at the templates to see what has changed if anything and comment further.
Sorry I did this a few days ago but got the comment in the wrong issue. Too many open windows.

**Comment by rinkug** [comment]
Thanks for such a detailed discussion folks.  After reading all the discussion items, it seems that we still have the six content types in the race with their original definitions.
(1) Event (or Announcements)
(2) Curated content
(3) Original article
(4) Blog article
(5) How-to's
(6) What-is

Are we in consensus with this, before we move to workflows? 

There is a lot of information in this discussion that would be useful in setting up the editorial process/workflows(such as  the ratings table: https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/issues/361#issuecomment-496261685)...




`

**Comment by markcmiller86** [comment]
@rinkug ... I can't recall but I don't think you were able to attend last week's telecon, right? There, we decided that types 3-6 should probably be handled under the broad heading of *Article* (my vote) or *Original Article* (@bernhold initial type name) as in [@bernhold comment above](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/issues/361#issuecomment-496261685). We decided I would submit a PR to document the contet types based on [@bernhold comment above](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/issues/361#issuecomment-496261685).

I submitted that PR as #373 and there has been some discussion there.
